### DISCUSSION TITLE
Allow for customization of files in app folder

### DISCUSSION THREAD

**Discussion by diderich** [YetiForceCompany/YetiForceCRM#16404] [Feature improvement]
Currently YetiForceCRM supports customizing files from the `layouts`, `modules`, and `languages` directories by copying (and changing) them in the `custom` folder. This is currently not possible/implemented for files from the `app` folder.

It would be great if the customization functionality could be extended to the `app` folder. The reasoning for customizing files from that folder lies in the possibility to customize the display/formatting/layout of the different data types, for example time formatting.

I have, for example, added a function `DateTime::formatToViewDateOnly($dateTime)` that only displays the date (without the time) and used that function in `custom/layouts/basic/modules/Vtiger/dashboards/CalendarActivitiesContents.tpl` to display the due date for Tasks and To Do item as follows
```
<div class="ml-1">
  <small>
    {if $ACTIVITY_TYPE eq 'Task' || $ACTIVITY_TYPE eq 'To Do'}
      {\App\Fields\DateTime::formatToViewDateOnly("$ACTIVITY_DATE $ACTIVITY_TIME")}
    {else}
      {\App\Fields\DateTime::formatToViewDate("$ACTIVITY_DATE $ACTIVITY_TIME")}
    {/if}
  </small>
</div>
```

**Comment by rskrzypczak** [comment]
You can always create a new helper class (file) in any module in the modules/ (custom/modules/) directory and refer to it wherever you need it.
Something like this file: https://github.com/YetiForceCompany/YetiForceCRM/blob/developer/modules/Vtiger/helpers/Util.php
This is a safer approach, because it will not overwrite your files when updating :)
